### PR TITLE
Feature/typescript 1

### PR DESCRIPTION
**Pull request by eklemen**
**What is this doing?**
Adding typescript as an option. Adds configs and types for both Sequelize and Mongoose/Mongo. 

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [] Yes
- [x] No

If yes, please describe how it impacts the application:
It should all still work (famous last words)

**Other information:**

### CODE REVIEW

**Comment by smaharj1** [comment]
Is there a typescript boilerplate that you are basing the starter project on?

**Comment by eklemen** [comment]
> Is there a typescript boilerplate that you are basing the starter project on?

@smaharj1 
Im basing it off of like 3-4 lol. Not sure what is considered best practice with typescript, so using Microsofts boilerplate as a base line

[Microsoft](https://github.com/microsoft/TypeScript-Node-Starter/tree/master/src)
[Some other repo](https://github.com/GeekyAnts/express-typescript)
[Not much from this one](https://github.com/jverhoelen/node-express-typescript-boilerplate)
[And this blog post](https://levelup.gitconnected.com/setup-express-with-typescript-in-3-easy-steps-484772062e01)

Part of why i opened this as a draft is because this is more of the foundational stuff with the generator, and just creating the files and a logical split between answering JS or TS in the prompts. Need some insight to the typescript itself.

**Comment by smaharj1** [comment]
Ah ok. Looks good. Following Msft sounds good

File: bin/express-cli.js [hunk 1]
```diff
@@ -116,52 +132,30 @@ inquirer
...
-          '@babel/node': '^7.8.7',
-          '@babel/preset-env': '^7.9.0',
-          'jest': '^25.2.7',
-          'npm-run-all': '^4.1.5',
-          'rimraf': '^3.0.2',
-          'nodemon': '^2.0.3'
-        }
+      const pkg = { ...codeSnippets.pkg }
+      pkg.name = kebabCase(name)
+      if (hasTs) {
```
Comment (smaharj1) [inline bin/express-cli.js:162]: We probably should move this logic to a separate file. its getting cluttered.

**Comment by eklemen** [comment]
@smaharj1 the [mongojs](https://www.npmjs.com/package/mongojs) package surprisingly does not have an associated `@types/` download. 

The official [mongodb](https://www.npmjs.com/package/mongodb) package, however, does.
Thoughts?
Edit: Looks like someone opened an [issue for mongojs](https://github.com/mongo-js/mongojs/issues/281) back in 2016 to add TS definitions. Someone made a gist as a starter then they lost the gist 🤦 . Either way doesn't seem like its maintained enough to warrant using it. 
Looks like we can probably just drop mongojs support and swap to mongodb. Will just be a separate PR.

File: bin/express-cli.js (hunk 1 above)
Comment (eklemen) [reply bin/express-cli.js:162]: Once we get this implemented, we can do some more cleanup. I'm all for organizing and beautifying it

**Comment by eklemen** [comment]
#### Typescript selected (removed mongojs)
ts-db

#### Javascript selected (options include mongojs)
image

**Comment by eklemen** [comment]
Will update the readme later to mention TS support.

**Comment by smaharj1** [comment]
Nice! It works great!

**Review by smaharj1 — APPROVED** [review]

**Pull request merged**